### ISSUE TITLE
loss nan even turn on ecc and use full precision

### ISSUE THREAD

**Issue by wasd2002** [d8ahazard/sd_dreambooth_extension#413]
problem engaged since yesterday webui update
corrupt seems cause randomly, when start over with the same parameters problem will not occur again
Dreambooth c2fc683ebcf4db23cfce22626aa6a7b565994c96
WebUI  46b0d230e7c13e247eabb22e1103ce512e7ed6b1

E:\sdwebui\venv\lib\site-packages\diffusers\utils\deprecation_utils.py:35: FutureWarning: The configuration file of this scheduler: DDIMScheduler {
  "_class_name": "DDIMScheduler",
  "_diffusers_version": "0.9.0",
  "beta_end": 0.012,
  "beta_schedule": "scaled_linear",
  "beta_start": 0.00085,
  "clip_sample": false,
  "num_train_timesteps": 1000,
  "prediction_type": "epsilon",
  "set_alpha_to_one": false,
  "steps_offset": 0,
  "trained_betas": null
}
 is outdated. `steps_offset` should be set to 1 instead of 0. Please make sure to update the config accordingly as leaving `steps_offset` might led to incorrect results in future versions. If you have downloaded this checkpoint from the Hugging Face Hub, it would be very nice if you could open a Pull request for the `scheduler/scheduler_config.json` file
  warnings.warn(warning + message, FutureWarning)
 Restored system models.
 Allocated: 6.0GB
 Reserved: 6.1GB

 Allocated 6.0/8.0GB
 Reserved: 6.1/8.1GB

Concept 0 class dir is C:\Users\User\Desktop\dbttclass
Starting Dreambooth training...
 Cleanup completed.
 Allocated: 0.0GB
 Reserved: 0.0GB

 Allocated 0.0/2.0GB
 Reserved: 0.0/3.3GB

Initializing dreambooth training...
Patching transformers to fix kwargs errors.
Replace CrossAttention.forward to use default
 Cleanup completed.
 Allocated: 0.0GB
 Reserved: 0.0GB

 Loaded model.
 Allocated: 0.0GB
 Reserved: 0.0GB

===================================BUG REPORT===================================
Welcome to bitsandbytes. For bug reports, please submit your error trace to: https://github.com/TimDettmers/bitsandbytes/issues
For effortless bug reporting copy-paste your error into this form: https://docs.google.com/forms/d/e/1FAIpQLScPB8emS3Thkp66nvqwmjTEgxp8Y9ufuWTzFyr9kJ5AoI47dQ/viewform?usp=sf_link
================================================================================
CUDA SETUP: Loading binary E:\sdwebui\venv\lib\site-packages\bitsandbytes\libbitsandbytes_cuda116.dll...
E:\sdwebui\venv\lib\site-packages\diffusers\utils\deprecation_utils.py:35: FutureWarning: It is deprecated to pass a pretrained model name or path to `from_config`.If you were trying to load a scheduler, please use <class 'diffusers.schedulers.scheduling_ddpm.DDPMScheduler'>.from_pretrained(...) instead. Otherwise, please make sure to pass a configuration dictionary instead. This functionality will be removed in v1.0.0.
  warnings.warn(warning + message, FutureWarning)
 Cleanup completed.
 Allocated: 0.8GB
 Reserved: 0.8GB

 Scheduler, EMA Loaded.
 Allocated: 7.2GB
 Reserved: 7.3GB

***** Running training *****
  Num examples = 178
  Num batches each epoch = 45
  Num Epochs = 267
  Instantaneous batch size per device = 4
  Total train batch size (w. parallel, distributed & accumulation) = 4
  Gradient Accumulation steps = 1
  Total optimization steps = 12000
   Training settings: CPU: False Adam: True, Prec: no, Grad: True, TextTr: False EM: True, LR: 4.28e-06
 Allocated: 7.2GB
 Reserved: 7.3GB

Steps:   0%|                              | 5/12000 [00:10<5:01:23,  1.51s/it, loss=0.102, lr=2.85e-9, vram=8.9/20.5GB] Step 5 completed.
 Allocated: 8.9GB
 Reserved: 20.5GB

Steps:   8%|██▎                        | 1000/12000 [16:58<3:05:01,  1.01s/it, loss=0.133, lr=2.83e-6, vram=8.9/20.5GB]Successfully trained model for a total of 1000 steps, converting to ckpt.
Compiling checkpoint: a2
[*] Weights saved at E:\sdwebui\models\dreambooth\a2
 Cleanup completed.
 Allocated: 8.9GB
 Reserved: 20.5GB

Steps:  17%|████▋                       | 2000/12000 [33:57<2:49:28,  1.02s/it, loss=0.13, lr=4.08e-6, vram=8.9/20.5GB]Successfully trained model for a total of 2000 steps, converting to ckpt.
Compiling checkpoint: a2
[*] Weights saved at E:\sdwebui\models\dreambooth\a2
 Cleanup completed.
 Allocated: 8.9GB
 Reserved: 20.5GB

Steps:  25%|███████                     | 3000/12000 [50:57<2:29:49,  1.00it/s, loss=0.13, lr=3.69e-6, vram=8.9/20.5GB]Successfully trained model for a total of 3000 steps, converting to ckpt.
Compiling checkpoint: a2
[*] Weights saved at E:\sdwebui\models\dreambooth\a2
 Cleanup completed.
 Allocated: 8.9GB
 Reserved: 20.5GB

Steps:  31%|████████▍                  | 3749/12000 [1:03:46<2:28:17,  1.08s/it, loss=nan, lr=3.39e-6, vram=8.9/20.5GB]Successfully trained model for a total of 3749 steps, converting to ckpt.
Compiling checkpoint: a2
Generating samples: 100%|████████████████████████████████████████████████████████████████| 1/1 [00:04<00:00,  4.60s/it]
[*] Weights saved at E:\sdwebui\models\dreambooth\a2█████████████████████████████████████| 1/1 [00:04<00:00,  4.60s/it]
 Cleanup completed.
 Allocated: 8.9GB
 Reserved: 20.5GB

Training complete.
 CLEANUP:
 Allocated: 8.9GB
 Reserved: 20.5GB

**Comment by d8ahazard** [comment]
How many images are you training with?

**Comment by wasd2002** [comment]
> How many images are you training with?

178 images

**Comment by d8ahazard** [comment]
Can you kindly try with the latest version? I've bumped the diffusers version and implemented multiple other fixes.

**Comment by wasd2002** [comment]
> Can you kindly try with the latest version? I've bumped the diffusers version and implemented multiple other fixes.

Seems fixed.

**Comment by 78Alpha** [comment]
Appears to have returned with SDXL training.
